AutoGear.lua

```diff
@@ -57,6 +57,20 @@ local TOC_VERSION_BFA = 80000
 local TOC_VERSION_SL = 90000
 local TOC_VERSION_DF = 100000
 
+-- store true or false for each expansion
+local isClassic = TOC_VERSION_CURRENT < TOC_VERSION_TBC
+local isTBC = TOC_VERSION_CURRENT >= TOC_VERSION_TBC and TOC_VERSION_CURRENT < TOC_VERSION_WOTLK
+local isWrath = TOC_VERSION_CURRENT >= TOC_VERSION_WOTLK and TOC_VERSION_CURRENT < TOC_VERSION_CATA
+local isCata = TOC_VERSION_CURRENT >= TOC_VERSION_CATA and TOC_VERSION_CURRENT < TOC_VERSION_MOP
+local isMOP = TOC_VERSION_CURRENT >= TOC_VERSION_MOP and TOC_VERSION_CURRENT < TOC_VERSION_WOD
+local isWOD = TOC_VERSION_CURRENT >= TOC_VERSION_WOD and TOC_VERSION_CURRENT < TOC_VERSION_LEGION
+local isLegion = TOC_VERSION_CURRENT >= TOC_VERSION_LEGION and TOC_VERSION_CURRENT < TOC_VERSION_BFA
+local isBFA = TOC_VERSION_CURRENT >= TOC_VERSION_BFA and TOC_VERSION_CURRENT < TOC_VERSION_SL
+local isSL = TOC_VERSION_CURRENT >= TOC_VERSION_SL and TOC_VERSION_CURRENT < TOC_VERSION_DF
+local isDF = TOC_VERSION_CURRENT >= TOC_VERSION_DF
+
+
+
```

Comment:
I would like to avoid the maintenance burden this system implies.  If we use any of these booleans, behavior changes based on TOC instead of function interfaces.  In the case you want to use `isCata` here for, we should do something more dynamic like checking the number of return values or the return type of `GetTalentTabInfo` instead.  That way, it won't stop working in the future, like a "Classic MoP", for example.

Reply:
dropping the isExpansion